invoke.ts

```diff
@@ -63,38 +63,14 @@ async function main() {
       break;
     }
     case "SUCCESS": {
-      // parse and print the response (assuming it is a vec)
-      // TODO: Move this scval serializing stuff to stellar-base
-      if (!response.resultXdr) {
-          throw new Error(`No result XDR: ${JSON.stringify(response)}`);
+      if (!response.resultMetaXdr) {
+        throw new Error(`No result meta XDR: ${JSON.stringify(response)}`);
       }
-      const result = xdr.TransactionResult.fromXDR(response.resultXdr, "base64");
-      const scvals = result.result().results()[0].tr().invokeHostFunctionResult().success();
 
-      if (scvals.length !== 1) {
-        throw new Error(`Invalid tx success response for invoke host, expected one host function scval`);
-      }
-      const scval = scvals[0];
-      // Hacky result parsing. We should have some helpers from the
-      // js-stellar-base, or the generated Typescript bindings.
-      let parsed: number | object | null = null;
-      switch (scval.switch()) {
-      case xdr.ScValType.scvU32(): {
-        parsed = scval.u32();
-        break;
-      }
-      case xdr.ScValType.scvI32(): {
-        parsed = scval.i32();
-        break;
-      }
-      case xdr.ScValType.scvVec(): {
-        // Total hack, we just assume the object is a vec. Good enough for now.
-        parsed = scval.vec()!.map(v => v.sym().toString());
-        break;
-      }
-      default:
-        throw new Error(`Unexpected scval type: ${scval.switch().name}`);
-      }
+      const result = xdr.TransactionResultMeta.fromXDR(response.resultMetaXdr, "base64");
```

Comment:
this is untested, so a sanity-check here would be great: `resultMetaXdr` is an instance of `TransactionResultMeta`, correct?

Reply:
@sreuland helpfully pointed me to [invokehostfunction_test.go](https://github.com/stellar/go/blob/soroban-xdr-next/services/horizon/internal/integration/invokehostfunction_test.go#L219-L220), which shows me that it's a `TransactionMeta`

Reply:
could verify at runtime locally, to test the test, it's a two-step, build the image with component versions, then run it, this is example using versions I grabbed as of today, since it's still kinda pre-release on some stuff:

```
system-test$ make CORE_GIT_REF=https://github.com/stellar/stellar-core.git#875f47e247cefc25c8a4b3982ee65610e6a620e3 \
       CORE_COMPILE_CONFIGURE_FLAGS="--disable-tests --enable-next-protocol-version-unsafe-for-production" \
       SOROBAN_RPC_GIT_REF=https://github.com/stellar/soroban-tools.git#main \
       RUST_TOOLCHAIN_VERSION=stable \
       SOROBAN_CLI_GIT_REF=https://github.com/stellar/soroban-tools.git#main \
       SYSTEM_TEST_IMAGE=stellar/system-test:dev \
       QUICKSTART_GIT_REF=master \
       JS_SOROBAN_CLIENT_NPM_VERSION=0.9.0 build

system-test$ docker run --rm -t --name e2e_test stellar/system-test:dev --SorobanExamplesGitHash main

```